### PR TITLE
Enable basic security/mTLS for consul

### PR DESCRIPTION
**Pull request by kidiyoor**
As discussed in - https://goo.gl/Dt11Ct and [https://github.com/istio/issues/issues/118](https://github.com/istio/issues/issues/118)

Enabling basic security for consul services, assuming all the consul services run in default service account.

### CODE REVIEW

**Comment by rshriram** [comment]
Sweet! Cc @GregHanson

File: pilot/pkg/serviceregistry/consul/controller.go [hunk 1]
```diff
@@ -198,5 +198,14 @@ func (c *Controller) AppendInstanceHandler(f func(*model.ServiceInstance, model.
 
 // GetIstioServiceAccounts implements model.ServiceAccounts operation TODO
 func (c *Controller) GetIstioServiceAccounts(hostname string, ports []string) []string {
-	return nil
+	// Need to get service account of service registered with consul
+	// Currently Consul do not have service account or equivalent concept
```
Comment (mandarjog) [inline pilot/pkg/serviceregistry/consul/controller.go]: nit: Consul **does** not have

File: pilot/pkg/serviceregistry/consul/conversion.go [hunk 2]
```diff
@@ -23,6 +23,8 @@ import (
 
 	"istio.io/istio/pilot/pkg/model"
 	"istio.io/istio/pkg/log"
+
+        meshconfig "istio.io/api/mesh/v1alpha1"
```
Comment (GregHanson) [inline pilot/pkg/serviceregistry/consul/conversion.go]: file needs to have `goimports` run

File: pilot/pkg/serviceregistry/consul/conversion.go [hunk 3]
```diff
@@ -147,3 +150,13 @@ func convertProtocol(name string) model.Protocol {
...
 }
+
+// Extracts security option for given port from labels. If there is no such
+// annotation, or the annotation value is not recognized, returns
+// meshconfig.AuthenticationPolicy_INHERIT
+func extractAuthenticationPolicy(port int, name string) meshconfig.AuthenticationPolicy {
+	// TODO: https://github.com/istio/istio/issues/3338
+	// Check for the label - auth.istio.io/<port> and return auth policy respectively
+
+	return meshconfig.AuthenticationPolicy_INHERIT
```
Comment (GregHanson) [inline pilot/pkg/serviceregistry/consul/conversion.go]: please add a testcase to conversion_test.go in this package for this case

**Comment by kidiyoor** [comment]
@GregHanson I haven't tried out the tag format - `auth.istio.io/8000: NONE` for consul. 

I have raised an issue - https://github.com/istio/istio/issues/3338 to track and discuss further on how the consul label for enabling/disabling per service mTLS should look.

File: pilot/pkg/serviceregistry/consul/conversion.go (hunk 3 above)
Comment (kidiyoor) [reply pilot/pkg/serviceregistry/consul/conversion.go]: adding simple test without consul labels for now.

File: pilot/pkg/serviceregistry/consul/conversion.go (hunk 2 above)
Comment (kidiyoor) [reply pilot/pkg/serviceregistry/consul/conversion.go]: not sure what this means. maybe because I'm very new to golang.

**Comment by myidpt** [comment]
@rshriram Could you add /lgtm if this PR looks good?

**Pull request merged**